code/osapi/osapi.cpp

```diff
@@ -512,21 +512,28 @@ bool os_is_legacy_mode()
 					<< "cmdline_fso.cfg";
 		old_config_time = std::max(old_config_time, get_file_modification_time(path_stream.str()));
 #else
-		// At this point we can't determine if the old config exists so just assume that it does
-		auto old_config_exists = true;
-		time_t old_config_time = os_registry_get_last_modification_time();
+		tl::optional<time_t> old_config_time = os_registry_get_last_modification_time();
```

Comment:
I'm not a fan of having `old_config_time` be two different types in two different compilation paths.  I recommend using similar code for both: make `old_config_time` `time_t` and initialize it to 0, and only set its value if `old_config_exists` is true.  You'll need to use a temporary `t1::optional` for the return value of the modification time.

So for example:
```C++
tl::optional<time_t> optional_old_config_time = os_registry_get_last_modification_time();
//Check if the registry key exists
auto old_config_exists = optional_old_config_time.has_value();
time_t old_config_time = 0;
if (old_config_exists) {
	old_config_time = old_config_time.value();
	[etc.]
```

Reply:
Done.
Also #6392 can be closed with this.